pkg/controller/core.oam.dev/v1alpha2/applicationcontext/applicationcontext_controller.go

```diff
@@ -107,9 +107,9 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 	// call into the old ac Reconciler and copy the status back
 	acReconciler := ac.NewReconciler(r.mgr, dm, r.log, ac.WithRecorder(r.record), ac.WithApplyOnceOnlyMode(r.applyMode))
 	reconResult := acReconciler.ACReconcile(ctx, appConfig, r.log)
+	appContextPatch := client.MergeFrom(appContext.DeepCopy())
 	appContext.Status = appConfig.Status
-	// always update ac status and set the error
-	err = errors.Wrap(r.client.Status().Update(ctx, appContext), errUpdateAppContextStatus)
+	err = errors.Wrap(r.client.Status().Patch(ctx, appContext, appContextPatch), errUpdateAppContextStatus)
```

Comment:
```suggestion
         // always update ac status and set the error
         // this should be the only place for status of AppContext to update, so we can patch to avoid update conflicts caused by `resourceVersion` changed by spec.
	err = errors.Wrap(r.client.Status().Patch(ctx, appContext, appContextPatch), errUpdateAppContextStatus)
```

Reply:
Let's keep the old commnets and add new to explain this behavior